src/test/java/org/opensearch/sdk/TestSDKRestRequest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.sdk;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Test;
+import org.opensearch.common.bytes.BytesArray;
+import org.opensearch.common.bytes.BytesReference;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.extensions.rest.ExtensionRestRequest;
+import org.opensearch.http.HttpRequest;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestRequest.Method;
+import org.opensearch.sdk.rest.SDKHttpRequest;
+import org.opensearch.sdk.rest.SDKRestRequest;
+import org.opensearch.test.OpenSearchTestCase;
+
+import static java.util.Map.entry;
+
+public class TestSDKRestRequest extends OpenSearchTestCase {
+    @Test
+    public void TestSDKRestRequestMethods() {
+        RestRequest.Method expectedMethod = Method.GET;
+        String expectedUri = "foobar?foo=bar&baz=42";
+        String expectedPath = "foo";
+        Map<String, String> expectedParams = Map.ofEntries(entry("foo", "bar"), entry("baz", "42"));
+        Map<String, List<String>> expectedHeaders = Map.ofEntries(entry("foo", Arrays.asList("hello", "world")));
+        XContentType exptectedXContentType = null;
```

Comment:
We should be able to test this better: if we pass as the headers a "Content-Type" of "application/json" then the `RestRequest` constructor does this:
```java
        final XContentType xContentType;
        try {
            xContentType = parseContentType(headers.get("Content-Type"));
        } catch (final IllegalArgumentException e) {
            throw new ContentTypeHeaderException(e);
        }
```

This also means that we don't really need to pass the xcontent type as part of the ExtensionRestRequest any more, as long as we're passing the XContent header.

Reply:
> Should I remove xcontent type from ExtensionRestRequest?

I'm ok with leaving it for now, even if we don't use it.  You might add a comment that it's not needed if we are sure the content header is passed, and It's a good "backup" if we somehow lose the header.  We can make the final call when @saratvemulapalli moves those requests to protobuf.